feat(auth): add role-based auth for categories (#34 #41)

infinity74853 opened the pull request:
- Create roles guard/decorator per NestJS docs
- Protect categories POST/PATCH/DELETE endpoints
- Only ADMIN can manage categories
- Fix DTOs and service types

infinity74853 committed:
feat(auth): add role-based auth for categories (#34 #41)

- Create roles guard/decorator per NestJS docs
- Protect categories POST/PATCH/DELETE endpoints
- Only ADMIN can manage categories
- Fix DTOs and service types

infinity74853 committed:
Merge remote-tracking branch 'upstream/week2' into week2-infinity74853

RolandSallaz commented on src/categories/entities/category.entity.ts:
```diff
@@ -12,9 +18,11 @@ export class Category {
   parent: Category | null;
 
-  @OneToMany(() => Category, category => category.children)
+  @OneToMany(() => Category, (category) => category.children)
```
Здесь мы связываем по полю category.parent

infinity74853 replied:
Исправлено.

RolandSallaz commented on src/auth/guards/roles.guard.ts:
```diff
@@ -0,0 +1,69 @@
+}
+
+// Интерфейс для Request
+interface RequestWithUser {
```
В auth/types.ts уже есть типы JwtPayload(зашитые в токен данные пользователя) и AuthenticatedRequest(данные пользователя в запросе), следует использовать их

infinity74853 replied:
Исправлено.

infinity74853 committed:
fix: resolve merge conflicts and update role-based auth

- Fix category entity relationships with proper parent/child references
- Update CreateCategoryDto with full validation and parentId support
- Integrate TypeORM repository in CategoriesService with business logic
- Use AuthenticatedRequest type from auth/types.ts in RolesGuard
- Add support for both role: string and roles: string[] formats
- Preserve role-based authorization for categories endpoints

infinity74853 committed:
fix: resolve merge conflicts and address review comments

- Fix category entity relationships as per review
- Use AuthenticatedRequest from auth/types.ts in RolesGuard
- Preserve role-based authorization for categories endpoints

infinity74853 commented:
Исправлены замечания и конфликты.
